src/DotLiquid/Template.cs

```diff
@@ -56,8 +56,8 @@ public class Template
         /// </summary>
         public static TimeSpan RegexTimeOut { get; set; }
 
-        private static readonly Dictionary<Type, Func<object, object>> SafeTypeTransformers;
-        private static readonly Dictionary<Type, Func<object, object>> ValueTypeTransformers;
+        private static readonly ConcurrentDictionary<Type, Func<object, object>> SafeTypeTransformers;

```

Comment:
Wouldn't this need to be accompanied by changes to adopt the thread safe dictionary methods per https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/how-to-add-and-remove-items
?

Reply:
And would need accompanying tests to prove it works it resolved the issue.

Reply:
Hi @daviburg, Updated the changes. Please check once.

Reply:
I see that you are using the proper method to update the concurrent dictionary now, but I do not yet understand how the conflict resolution logic avoids prior race conditions.